src/chunk_cache.rs

```diff
@@ -359,11 +384,16 @@ impl SimpleDiskCache {
     async fn remove(&self, key: &str) {
         let mut state = self.load_state().await;
         if let Some(data) = state.metadata.remove(key) {
-            let path = self
-                .dir
-                .join(&self.name)
-                .join(self.filename_for_key(key).await);
-            fs::remove_file(path).await.unwrap();
+            let path = self.dir.join(&self.name);
+            // Remove the chunk file
+            fs::remove_file(path.join(self.filename_for_key(key).await))
+                .await
+                .unwrap();
+            // Remove the metadata file
+            fs::remove_file(path.join(self.filename_for_key(key).await + ".meta"))
+                .await
+                .unwrap();
```

Comment:
Is there anything we can do to avoid the unwraps here? Or we could at least log the error for easier debugging with something like
```suggestion
            fs::remove_file(path.join(self.filename_for_key(key).await))
                .await
                .map_err(|e| log::error!("Failed to remove cache data with: {}", e)
                .unwrap();
            // Remove the metadata file
            fs::remove_file(path.join(self.filename_for_key(key).await + ".meta"))
                .await
                .map_err(|e| log::error!("Failed to remove cache metadata with: {}", e)
                .unwrap();
```

N.B. We might also want to add this `map_error` pattern anywhere else in the code that we need to use unwrap / expect.